packages/core/factories/src/useUserFactory.ts

```diff
@@ -0,0 +1,113 @@
+import { ref, Ref, watch, computed } from '@vue/composition-api';
+import { UseUser } from '@vue-storefront/interfaces';
+
+export type UseUserFactoryParams<USER, CART, UPDATE_USER_PARAMS> = {
+  cart: Ref<CART>;
+  loadUser: () => Promise<USER>;
+  logOut: (params?: {currentUser?: USER}) => Promise<void>;
+  updateUser: (params: {currentUser: USER; paramsToUpdate: UPDATE_USER_PARAMS}) => Promise<USER>;
```

Comment:
shouldn't that be always an updated `currentUser` ? is there a use case when we selectively send only a subset of form elements?

Reply:
if I understand correctly, you want to keep it always like that?
```suggestion
updateUser: (params: {currentUser?: USER; paramsToUpdate: UPDATE_USER_PARAMS}) => Promise<USER>;
```

Reply:
not  entirely. What i meant is `paramsToUpdate` is implying that we are passing only a subset of `currentUser` data that we want to update while in most cases when you save a form you're sending all the values, not only the updated ones therefore it should be something like `updatedUserData` or similar

Reply:
I believe what Filip was in mind:
```suggestion
  updateUser: (params: {currentUser: USER}) => Promise<USER>;
```

Reply:
Now it's clear thanks

Reply:
no it's not @patzick  :D

Reply:
```suggestion
  updateUser: (params: {currentUser: USER; updatedUserData: UPDATE_USER_PARAMS}) => Promise<USER>;
```

Reply:
Done